docs/features/walletconnect.md

```diff
@@ -0,0 +1,106 @@
+
+Information exchange between the peers typically consists of an unsigned atomic transaction group from the dApp to the mobile wallet, which is then inspected by the user, signed and returned back to the dApp. The dApp will submit the signed object for confirmation by the Algorand network which will effect balances and state accordingly.
+
+<center>![WalletConnect Header](../imgs/walletconnect-establishing-connection.png)</center>
+
+## Quick start for dApp devs
+
+The TypeScript sample code below should allow a dApp developer to establish a connection and request a simple payment transaction for signature. A [detailed example](https://github.com/jasonpaulos/algorand-walletconnect-example-dapp) is offered in React and a [live demo](https://jasonpaulos.github.io/algorand-walletconnect-example-dapp/) is available on TestNet.
+
+### Install
+
+```bash tab="yarn"
+yarn add @walletconnect/client @walletconnect/qrcode-modal @algorand/js-algorand-sdk
+```
+
+```bash tab="npm"
+npm install --save @walletconnect/client @walletconnect/qrcode-modal @algorand/js-algorand-sdk
+```
+
+### Initiate Connection
+
+```typescript
+import WalletConnect from "@walletconnect/client";
+import QRCodeModal from "@walletconnect/qrcode-modal";
+import algosdk from "@algorand/js-algorand-sdk";
+
+// Create a connector
+const connector = new WalletConnect({
+  bridge: "https://bridge.walletconnect.org", // Required
+  qrcodeModal: QRCodeModal,
+});
+
+// Check if connection is already established
+if (!connector.connected) {
+  // create new session
+  connector.createSession();
+}
+
+// Subscribe to connection events
+connector.on("connect", (error, payload) => {
+  if (error) {
+    throw error;
+  }
+
+  // Get provided accounts and chainId
+  const { accounts, chainId } = payload.params[0];
+});
+
+connector.on("session_update", (error, payload) => {
+  if (error) {
+    throw error;
+  }
+
+  // Get updated accounts and chainId
+  const { accounts, chainId } = payload.params[0];
+});
+
+connector.on("disconnect", (error, payload) => {
+  if (error) {
+    throw error;
+  }
+
+  // Delete connector
+});
+```
+
+### Sign Transaction
+
+```typescript
+// Draft transaction
+const txn = algosdk.makePaymentTxnWithSuggestedParamsFromObject({
+    from: "ALICEU3WMO5XYJVSODKJSYLFILIXXBEXHKIVSMX7GMGXJAYGFCJKVSQTUE",
+    to: "HZ57J3K46JIJXILONBBZOHX6BKPXEM2VVXNRFSUED6DKFD5ZD24PMJ3MVA",
+    amount: 100000,
+    suggestedParams,
+});
+
+// Sign transaction
+connector
+  .signTransaction(txn)
```

Comment:
Unfortunately it's not this simple...

See https://github.com/jasonpaulos/algorand-walletconnect-example/blob/d5c4fa4b333374ebbb757e10f1899d8567ac3367/src/server/server.ts#L93-L128 for a complete example. The relevant parts are:

```javascript
    // txns is an array of algosdk.Transaction
    const txnsToSign = txns.map(txn => {
      const encodedTxn = Buffer.from(algosdk.encodeUnsignedTransaction(txn.txn)).toString("base64");

      return {
        txn: encodedTxn,
        message: 'Description of transaction being signed',
        // Note: if the transaction does not need to be signed (because it's part of an atomic group
        // that will be signed by another party), specify an empty singers array like so:
        // signers: [],
      };
    });

    const requestParams = [txnsToSign];

    const request = formatJsonRpcRequest(ALGORAND_SIGN_TRANSACTION_REQUEST, requestParams);
    const result: Array<string | null> = await this.connector.sendCustomRequest(request);
    const decodedResult = result.map(element => {
      return element ? new Uint8Array(Buffer.from(element, "base64")) : null;
    });

    return decodedResult;
```

This is currently messy, and it will be improved soon, tracked by https://github.com/algorand/js-algorand-sdk/issues/411.